spec/lib/query/query_donations_spec.rb

```diff
@@ -72,6 +72,69 @@
       export = vector_to_hash(campaign_export)
       expect(export.map{|i| i['Campaign Creator Email']}).to match_array([profile_email, profile_email, profile_email1, profile_email2])
     end
+
+    # we need to remove this context when we remove the address attributes from supporters table
+    context 'when looking at the supporter address' do
+      before do
+        supporter.save!
+
+        # Update directly on the database to avoid updating primary_address
+        Qx.update(:supporters)
+          .set(address: 'Some street', city: 'Aguas Claras', country: 'Brazil', state_code: 'DF', zip_code: '4002-8922')
+          .where(id: supporter.id)
+          .execute
```

Comment:
Avoid using `Qx` for new code. A better way of handling that would be as follows:

```ruby
Supporter.where(id: supporter.id).update_all(address: 'Some street', city: 'Aguas Claras', country: 'Brazil', state_code: 'DF', zip_code: '4002-8922')
```

[`update_all`](https://api.rubyonrails.org/v4.2/classes/ActiveRecord/Relation.html#method-i-update_all) skips validations, callbacks and and type-casting and goes directly to the database.